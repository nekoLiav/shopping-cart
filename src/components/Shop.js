import { useState, useEffect } from 'react';
import CartSummary from './CartSummary';
import ShopItems from './ShopItems';
import ShoppingCart from './ShoppingCart';

const Shop = () => {
  const [shopItems, setShopItems] = useState([]);
  const [cart, setCart] = useState([]);
  const [cartView, setCartView] = useState(false);

  useEffect(() => {
    // setShopItems(shopData.map((item) => Object.assign(item, { quantity: 1 })));
    const fakeStoreAPIFetch = async () => {
      const fetchedData = await fetch('https://fakestoreapi.com/products/');
      const response = await fetchedData.json();
      console.log(response);
      setShopItems(response.map((items) => Object.assign(items, { quantity: 1 })));
    };
    fakeStoreAPIFetch();
    console.log(shopItems);
  }, []);

  const handleQuantityChange = (item, e) => {
    setShopItems((items) =>
      items.map((items) =>
        items.title === item.title ? { ...items, quantity: Number(e.target.value) } : items,
      ),
    );
  };

  const addToCart = (item) => {
    if (cart.map((items) => items.title).includes(item.title)) {
      setCart((items) =>
        items.map((items) => {
          if (items.title === item.title) {
            return {
              ...items,
              price: item.price * (items.quantity + item.quantity),
              quantity: items.quantity + item.quantity,
            };
          } else {
            return items;
          }
        }),
      );
    } else {
      setCart((items) => [...items, { ...item, price: item.price * item.quantity }]);
    }
  };

  return (
    <div className='shop'>
      <div className='shop-content'>
        <h1 className='shop-header'>Shop</h1>
        <ShopItems
          shopItems={shopItems}
          changeQuantity={handleQuantityChange}
          addToCart={addToCart}
        />
      </div>
      {cartView ? <ShoppingCart cart={cart} /> : null}
      <CartSummary cart={cart} toggleView={() => setCartView(cartView ? false : true)} />
    </div>
  );
};

export default Shop;
