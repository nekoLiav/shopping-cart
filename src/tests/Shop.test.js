import { render, screen } from '@testing-library/react';
import Shop from '../components/Shop';

describe('Shop component', () => {
  it('matches snapshot', () => {
    const { container } = render(<Shop />);
    expect(container).toMatchSnapshot();
  });
  it('renders correct heading', () => {
    const { getByRole } = render(<Shop />);
    expect(getByRole('heading', { name: 'Shop' }));
  });
});
