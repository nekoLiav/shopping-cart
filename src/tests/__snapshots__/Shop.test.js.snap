// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component matches snapshot 1`] = `
<div>
  <div
    class="shop"
  >
    <h1
      class="page-header-text"
    >
      Shop
    </h1>
    <div
      class="shop-content"
    >
      <div
        class="shop-items"
      />
    </div>
    <div
      class="cart-summary"
    >
      <p>
        View Cart | Your cart is empty.
      </p>
    </div>
  </div>
</div>
`;
